fakelibusb_devices.go

```diff
@@ -186,4 +186,43 @@ var fakeDevices = []*DeviceDesc{
 			}},
 		}},
 	},
+	// Bus 001 Device 004: ID 0333:0002
+	// One config, three interfaces, zero setups,
+	// zero endpoints.
+	// This was made to test detachKernelDriver
+	&DeviceDesc{
```

Comment:
if you split the "failing detach" to a separate test function, you don't need a new device.